tools/veneer-update.go

```diff
@@ -61,215 +46,110 @@ func main() {
 	}
 }
 
-func createPatchVersionVeneer(veneerFile, channel, currentVersion, newVersion string) ([]byte, error) {
-	source, err := os.ReadFile(veneerFile)
-	if err != nil {
-		return nil, fmt.Errorf("could not read veneer file: %w", err)
-	}
-
-	ret, err := updateVeneerFile(source, "olm.channel", "name", channel, func(m map[string]any) (map[string]any, error) {
-		entries, ok := m["entries"]
-		if !ok {
-			return nil, errors.New("could not find entries")
-		}
-
-		eSlice, ok := entries.([]any)
-		if !ok {
-			return nil, errors.New("could not assert to slice")
-		}
-
-		i, e, err := findEntry(eSlice, currentVersion)
-		if err != nil {
-			return nil, err
-		}
-
-		// Keep just the name config in the current version
-		eSlice[i] = map[string]any{"name": e["name"]}
-
-		// Create a copy of the current version and rename to new version
-		newEntry := e
-		newEntry["name"] = fmt.Sprintf("everest-operator.v%s", newVersion)
+type Entry struct {
+	Schema         string    `yaml:"schema"`
+	DefaultChannel string    `yaml:"defaultChannel,omitempty"`
+	Name           string    `yaml:"name,omitempty"`
+	Versions       []Version `yaml:"entries,omitempty"`
+	Package        string    `yaml:"package,omitempty"`
+	Image          string    `yaml:"image,omitempty"`
+}
 
-		// Add current version to skips
-		skips, ok := newEntry["skips"]
-		if !ok {
-			skips = []string{}
-		}
+type Template struct {
+	Schema  string  `yaml:"schema"`
+	Entries []Entry `yaml:"entries"`
+}
```

Comment:
Should we import the modules and use the official structs instead?
Your `Template` is the  [BasicTemplate](https://github.com/operator-framework/operator-registry/blob/3be1d97c5d8dfe672ba66e7b8fd4aaaa21fe7d2b/alpha/template/basic/basic.go#L21-L24) and the `Entry` is the [DeclarativeConfig and child structs](https://github.com/operator-framework/operator-registry/blob/3be1d97c5d8dfe672ba66e7b8fd4aaaa21fe7d2b/alpha/declcfg/declcfg.go#L25-L60)

Reply:
Thanks for highlighting this! Updated and refactored. 

Advantages:
1. Consistent structs. We know for sure that the structs we use reflect the basic template structure.

Disadvantages: 
1.  more code bc: 
- the structs doesn't have the yaml tags so we need to handle the json-yaml transformations 
- marshalling and unmarshalling blobs inside the structure
3. A big number of new indirect dependencies

In general the advantage may worth the disadvantages. 
Considering the refactoring, the current implementation looks solid and maintainable.